### DISCUSSION TITLE
Setting TMux Pane Title

### DISCUSSION THREAD

**Discussion by Yeraze** [JanDeDobbeleer/oh-my-posh#1844] [Troubleshoot]
Running into some issues getting on-my-posh with Fish to set my tmux pane titles.

I tried using the "console_title_template" features with tmux's support for escaped terminal commands and couldn't get it to work.  Something like:

```
  "console_title_template": "\u033k{{.Folder}}{{if .Root}} :: root{{end}} :: {{.Shell}}\u033",
```
Just generated errors..

So I tried using the type:command block to call a bash command of `tmux rename-window {{.Folder}}` and it seems that doesn't expand the variables.

Is there a way to get this to work?

I've tried `  "console_title_template": "\u033k{{.Folder}}{{if .Root}} :: root{{end}} :: {{.Shell}}\u033",` and get errors like

```
comments:3:29: invalid char escape
comments:3:29: invalid char escape
comments:3:29: invalid char escape
comments:3:29: invalid char escape
Oh My Posh Error:

…\u033k{{.Folder}|..., bigger context ...|emes/schema.json","console_title_template":"\u033k{{.Folder}}{{if .Root}} :: root{{end}} :: {{.Shell|...
```

I've tried
`  "console_title_template": "\033k{{.Folder}}{{if .Root}} :: root{{end}} :: {{.Shell}}\033",`
And I get a different error, but still an error:
```
 Oh My Posh Error:

…e":"\033k{{.Fold|..., bigger context ...|n/themes/schema.json","console_title_template":"\033k{{.Folder}}{{if .Root}} :: root{{end}} :: {{.Sh|...
```

I've tried `  "console_title_template": "\\033k{{.Folder}}{{if .Root}} :: root{{end}} :: {{.Shell}}\\033",` and it doesn't error, but it seems to just do nothing at all.

If I do a `echo \033kASDF\033` from the same terminal then it properly sets the pane title to `ASDF` so tmux seems to be configured properly.

**Comment by JanDeDobbeleer** [comment]
from what I gather you need to tell tmux to listen to the escape value: https://superuser.com/a/1098626

**Reply by Yeraze** [reply]
Look below but AFAICT tmux is respecting the escape codes, I just can't figure out how to get omp to print them.

**Comment by Yeraze** [comment]
Well how do I get oh-my-posh to actually output the escape value?  The `\u033` in the console_title_template just generates errors.. 

```
comments:3:29: invalid char escape
comments:3:29: invalid char escape
comments:3:29: invalid char escape
comments:3:29: invalid char escape
Oh My Posh Error:

…\u033k{{.Folder}|..., bigger context ...|emes/schema.json","console_title_template":"\u033k{{.Folder}}{{if .Root}} :: root{{end}} :: {{.Shell|...
```

If I do a `echo \033kASDF\033` from the same terminal then it properly sets the pane title to `ASDF` so tmux seems to be configured properly.

**Reply by Yeraze** [reply]
I've tried:

`  "console_title_template": "\u033k{{.Folder}}{{if .Root}} :: root{{end}} :: {{.Shell}}\u033",`
And get the error shown above..

I've tried
`  "console_title_template": "\033k{{.Folder}}{{if .Root}} :: root{{end}} :: {{.Shell}}\033",`
And I get a different error, but still an error:
```
 Oh My Posh Error:

…e":"\033k{{.Fold|..., bigger context ...|n/themes/schema.json","console_title_template":"\033k{{.Folder}}{{if .Root}} :: root{{end}} :: {{.Sh|...
```

I've tried `  "console_title_template": "\\033k{{.Folder}}{{if .Root}} :: root{{end}} :: {{.Shell}}\\033",` and it doesn't error, but it seems to just do nothing at all.

**Reply by JanDeDobbeleer** [reply]
It does that for you. You only need to worry about the text.

**Reply by Yeraze** [reply]
Yeah, it doesn't work.

Adding into my .tmux.conf the 2 commands above (Which I already had) just sets the Terminal title (using iTerm on Mac), but has no bearing on the Pane title.. Also, has nothing to do with the omp settings.. Which I'm trying to get terminal & folder..

**Reply by JanDeDobbeleer** [reply]
@Yeraze what you're trying to solve goes beyond the scope of oh-my-posh. We're still in the shell, not tmux as that's a terminal multiplexer.

**Reply by Yeraze** [reply]
Yeah I know.. I was trying to figure out how can I get omp to output Escape Sequences that tmux would interpret and handle, but it seems it can't be done?  I can enter escape sequences through the 'Command' segment, but that won't handle expansions... And the console_title ones support expansion, but not escape sequences.

**Reply by JanDeDobbeleer** [reply]
Pretty sure that even if expansion worked, it still wouldn't work as we're in a sub process and not the active shell session.
